airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/StagingFilenameGenerator.java

```diff
@@ -21,9 +22,18 @@
   // file suffix; its value range will be [1, maxPartsPerFile]
   private int currentFileSuffixPartCount = 0;
 
-  public StagingFilenameGenerator(final String streamName, final int maxPartsPerFile) {
+  // This variable is responsible to set the size of chunks size (In MB). After chunks created in
+  // S3 or GCS they will be uploaded to Snowflake or Redshift. These service have some limitations for the uploading file.
+  // So we make the calculation to determine how many parts we can put to the single chunk file.
+  private final long iterations;
+
+  /**
+   * @param streamName - the name of table will be processed
+   * @param chunkSize - the number of optimal chunk size for the service.
+   */
+  public StagingFilenameGenerator(final String streamName, final long chunkSize) {
     this.streamName = streamName;
-    this.maxPartsPerFile = maxPartsPerFile;
+    this.iterations = MAX_FILE_SIZE / chunkSize;
```

Comment:
sorry, I might be confused here - what was the error with `chunkSize / DEFAULT_MAX_BATCH_SIZE_BYTES`? My understanding is that this class is supposed to create files with at most `chunkSize` bytes

If I'm reading this correctly: `MAX_FILE_SIZE / chunkSize` means that the CopyConsumerFactory might generate chunks of variable size, and this class is targeting 15 GiB files?

Reply:
Not quite so, we introduce MAX_FILE_SIZE constant to determine some abstract possible size of file. After we split it on chunkSize to set approximately size of each chunk. So, the purpose of that formula to determine how many iterations we need to have a file with expected size. May be we can have some more optimal algorithm and formula for that flow. I hope I have described the idea behind that.

Reply:
BTW, I noticed that recommendation about snowflake here https://community.snowflake.com/s/article/faq-does-snowflake-have-a-recommended-file-size-for-loading
That is only related to **parallel** processing of file, and in code we have sequential processing https://github.com/airbytehq/airbyte/blob/25db8fe54cd96ff7deb4564d5ea936a79f3c9430/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/s3/LegacyS3StreamCopier.java#L158-L158

Reply:
chatted over a call - let's switch MAX_FILE_SIZE to 1 GiB so that we're targeting the same file size as Redshift.

I'll check with the team if the `copyS3CsvFileIntoTable` call should be done in parallel, and will make an issue if yes.